src/CommandLine/Text/HelpText.cs

```diff
@@ -17,8 +17,71 @@ namespace CommandLine.Text
     /// Provides means to format an help screen.
     /// You can assign it in place of a <see cref="System.String"/> instance.
     /// </summary>
+    
+    
+    
+    public struct ComparableOption
+    {
+        public bool Required;
+        public bool IsOption;
+        public bool IsValue;
+        public string LongName;
+        public string ShortName;
+        public int Index;
+    }
+    
     public class HelpText
     {
+
+
+        ComparableOption ToComparableOption(Specification spec, int index)
+        {
+            OptionSpecification option = spec as OptionSpecification;
+            ValueSpecification value = spec as ValueSpecification;
+            bool required = option != null ? option.Required : false;
+
+            return new ComparableOption()
+            {
+                Required = required,
+                IsOption = option != null,
+                IsValue = value != null,
+                LongName = option?.LongName,
+                ShortName = option?.ShortName,
+                Index = index
+            };
+        }
+
+
+        public Comparison<ComparableOption> OptionComparison = null;
+
+        public static Comparison<ComparableOption> RequiredThenAlphaComparison  = (ComparableOption attr1, ComparableOption attr2) =>

```

Comment:
Remove redundant else and braces, and use StringComparison.Ordinal

           public static Comparison<ComparableOption> RequiredThenAlphaComparison  = (attr1, attr2) =>
        {
            if (attr1.IsOption && attr2.IsOption)
            {
                if (attr1.Required && !attr2.Required)
                    return -1;

                if (!attr1.Required && attr2.Required)
                    return 1;

                int t = string.Compare(attr1.LongName, attr2.LongName, StringComparison.Ordinal);
                return t;
            }

            return attr1.IsOption && attr2.IsValue ? -1 : 1;
        };